### PR TITLE
feat(cdk/testing): add methods `getOptionalHarness` and `hasHarness`

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/cdk/testing/component-harness.ts [hunk 1]
```diff
@@ -403,10 +421,18 @@ export abstract class ContentContainerComponentHarness<S extends string = string
     return (await this.getRootHarnessLoader()).getHarness(query);
   }
 
+  async getOptionalHarness<T extends ComponentHarness>(query: HarnessQuery<T>): Promise<T | null> {
```
Comment (crisbeto) [inline src/cdk/testing/component-harness.ts:424]: Should it be called `getHarnessOptional` to align with the naming of other methods like `locatorForOptional`?

**Review by devversion — APPROVED** [review]
LGTM, agreeding with Kristiyan's comment but not feeling strongly.

File: src/cdk/testing/component-harness.ts [hunk 2]
```diff
@@ -96,13 +96,31 @@ export interface HarnessLoader {
...
    * @param query A query for a harness to create
    * @return A list instances of the given harness type.
    */
   getAllHarnesses<T extends ComponentHarness>(query: HarnessQuery<T>): Promise<T[]>;
+
+  /**
+   * Searches for an instance of the component corresponding to the given harness type under the
+   * `HarnessLoader`'s root element, and returns a boolean indicating if any were found.
+   * @param query A query for a harness to create
+   * @return An instance of the given harness type (or null if not found).
```
Comment (andrewseguin) [inline src/cdk/testing/component-harness.ts:121]: Update to reflect that it returns a boolean

**Review by andrewseguin — APPROVED** [review]
LGTM to me as well - also not sure if `getOptionalHarness` is the right name but not very opinionated on it.

File: src/cdk/testing/component-harness.ts (hunk 1 above)
Comment (mmalerba) [reply src/cdk/testing/component-harness.ts:424]: I don't think it makes to follow the same naming as `locatorForOptional`, in that case its an element that's optional so the name is short for `locatorForOptionalElement` which does line up with `getOptionalHarness`. For possible alternatives, what about: `getHarnessOrNull`, `maybeGetHarness`?

Comment (mmalerba) [reply src/cdk/testing/component-harness.ts]: I went with `getHarnessOrNull` based on a quick poll of some users

**Pull request merged**